app/javascript/components/bathroom.jsx

```diff
@@ -42,14 +38,14 @@ Bathroom.propTypes = {
   location: PropTypes.shape({
     bathroomCode: PropTypes.string,
   }),
-  loading: PropTypes.bool,
-  error: PropTypes.bool,
 };
 
 Bathroom.defaultProps = {
   location: {},
-  loading: true,
-  error: false,
 };
 
-export default withFragment(Bathroom, { location: getBathroomCode });
+export default compose(
+  renderWhileError(ErrorMessage),
+  renderWhileLoading(LoadingMessage),
+  withFragment({ location: getBathroomCode }),
+)(Bathroom);
```

Comment:
I'm not a fan of using the `compose` function here. It's not saving us much in the way of complexity or line count, but it is an extra third-party dependency and an extra concept the reader will have to understand.

```suggestion
const BathroomWithError = renderWhileError(Bathroom, ErrorMessage);
const BathroomWithErrorAndLoading = renderWhileLoading(BathroomWithError, LoadingMessage);
const BathroomWithErrorLoadingAndFragment = withFragment(BathroomWithErrorAndLoading, { location: getBathroomCode });

export default BathroomWithErrorLoadingAndFragment
```

Reply:
I disagree. It saves us from extra churn when want to inject/reorder HoCs e.g. if in your example we switch to loading preceding error I should rename three variables.